ui/dashboards/src/components/TimeRangeControls/TimeRangeControls.tsx

```diff
@@ -41,5 +43,26 @@ export function TimeRangeControls() {
       });
     }
   }
-  return <DateTimeRangePicker timeOptions={TIME_OPTIONS} value={timeRange} onChange={setTimeRange} />;
+
+  const handleRefresh = () => {
+    if (isRelativeTimeRange(timeRange)) {
+      setTimeRange({ ...timeRange, end: new Date() });
+    }
+  };
+
+  return (
+    <>
+      <DateTimeRangePicker timeOptions={TIME_OPTIONS} value={timeRange} onChange={setTimeRange} />
+      <RefreshIconButton aria-label="Refresh Dashboard" onClick={handleRefresh}>
+        <RefreshIcon />
+      </RefreshIconButton>
+    </>
+  );
```

Comment:
Do you think we should only show the button if `isRelativeTimeRange` is true? Or have it disabled and show a tooltip explaining that it can only be used for relative time ranges?

Might be a question for Mohit too

Reply:
Oh oops this says draft, I may have jumped the gun

Reply:
IMO I think refresh shouldn't be specific to relative time ranges. Refresh should just mean - "force queries to refetch data and recompute the time range". So even if you have an AbsoluteTimeRange, this should force queries to refetch. There are situations where queries fail, or load only partial data and want to refetch the query. 

Grafana does this as well, and I think it's a useful feature.

Reply:
yep i also talked to Mohit and he agreed that it's fine to let users refresh even if it's absolute time range